specutils/analysis/moment.py

```diff
@@ -43,10 +44,21 @@ def moment(spectrum, regions=None, order=0, axis=-1):
                                order=order, axis=axis)
 
 
-def _compute_moment(spectrum, regions=None, order=0, axis=-1):
+def _compute_moment(spectrum, regions=None, order=0, axis=None):
     """
     This is a helper function for the above `moment()` method.
     """
+    if axis is None:
+        if isinstance(spectrum, SpectrumCollection):
+            axes = [spec.spectral_axis_index for spec in spectrum]
+            if not np.all([x==axes[0] for x in axes]):
+                raise ValueError("All spectra in SpectrumCollection must have the same "
+                                 "spectral_axis_index for simultaneous moment calculation.")
+            axis = axes[0] + 1
```

Comment:
Why is this being done?  Isn't this setting the axis to be `axes[0].spectral_axis + 1`, which is not the spectral axis?  Or am I missing some logic?  (that happens easily during reviews...)

Reply:
`axis` is zero-indexed, right?  not 1-indexed?

Reply:
or is `spectral_axis_index` done in FITS convention and grabbed from the FITS header?  That seems like not what we want

Reply:
This is a great question, I'm going to have to think to remember what the heck I was thinking here - I coded this a while ago.

Reply:
Ah right, I remember now - it's because my understanding is that a `SpectrumCollection` adds a leading axis when it stacks the individual spectra together.

Reply:
Oh, that makes sense!  Add a comment about it, then?

Reply:
Done!